.github/workflows/2-find-commit-in-history.yml

```diff
@@ -1,81 +1,102 @@
-name: Step 2, Find a commit in history
+name: Step 2
 
-# This step listens for the learner creating or editing an issue comment.
-# This workflow updates from step 2 to step 3.
-
-# This will run every time we create or edit an issue comment.
-# Reference: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
 on:
-  workflow_dispatch:
   issue_comment:
     types: [created, edited]
 
 permissions:
-  # Need `contents: read` to checkout the repository.
-  # Need `contents: write` to update the step metadata.
   contents: write
+  actions: write
+  issues: write
+
+env:
+  STEP_3_FILE: ".github/steps/3-fix-broken-sidebar.md"
 
 jobs:
-  # Get the current step to only run the main job when the learner is on the same step.
-  get_current_step:
-    name: Check current step number
+  find_exercise:
+    name: Find Exercise Issue
+    uses: skills/exercise-toolkit/.github/workflows/find-exercise-issue.yml@v0.3.0
+
+  
+  validate_commit_reference:
+    name: Validate Commit Reference
     runs-on: ubuntu-latest
+    needs: [find_exercise]
+    outputs:
+      commit_id: ${{ steps.get_commit_id.outputs.commit_id }}
+
     steps:
-      - name: Checkout
+      - name: Checkout Repository
         uses: actions/checkout@v4
-      - id: get_step
-        run: |
-          echo "current_step=$(cat ./.github/steps/-step.txt)" >> $GITHUB_OUTPUT
+
       - id: get_commit_id
         run: |
-          echo "commit_id=$(cat ./.github/files/SIDEBARCOMMIT)" >> $GITHUB_OUTPUT
-    outputs:
-      current_step: ${{ steps.get_step.outputs.current_step }}
-      commit_id: ${{ steps.get_commit_id.outputs.commit_id }}
+          echo "commit_id=$(cat .github/files/SIDEBARCOMMIT)" >> $GITHUB_OUTPUT
 
-  on_fix_the_sidebar_issue_comment:
-    name: Check if the issue comment is referencing the correct commit ID
-    needs: get_current_step
+  check_step_work:
+    name: Check Step Work
+    runs-on: ubuntu-latest
+    needs: [find_exercise, validate_commit_reference]
+    if: contains(github.event.comment.body, ${{ needs.validate_commit_reference.outputs.commit_id }})
```

Comment:
We usually let the job run for wider conditions/without filters and check for what needed to be done within the job.

This gives you the option to provide feedback to the user, unlike here, they will not receive any feedback or progress in the exercise if they don't provide the commit_id, or in other cases `Duplicate of` or `Closes` comments.

Please take a look at https://github.com/skills/exercise-template/blob/main/.github/workflows/2-step.yml workflow's `check_step_work` and how it guides the user. 

I'm free to sync up and talk on this